### PR TITLE
Refactor checkout sidebar upsell for Jetpack 2 year plans.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR updates the checkout page to accommodate the Jetpack 2-year plans (Backup T1 Bi-Yearly & Security T1 Bi-Yearly).

Asana Task: 1201210512993648-as-1204659654942251/f

## Proposed Changes

* Makes the intro pricing text show in the variant dropdown item for 2 year plans.
* Makes the checkout sidebar billing-term upsell box display the correct discount percentage and better displays the comparison to make more sense.
* Some small CSS updates to accommodate these changes.

## Testing Instructions

- First, sandbox `public-api.wordpress.com`.
- In your sandbox, open file: `wp-content/admin-plugins/wpcom-billing/default-shopping-cart-json-serializer.php`.
- Comment out lines 1050 - 1052 and save file, so that we can view Jetpack **Bi-Yearly** products at checkout.  It should looks like this:
```
// if ( 'jetpack' === $product->product_type && $product->renews_biyearly() ) {
// 	return null;
// }
```
- Now, checkout this PR/branch and spin up Calypso: `git fetch && git checkout update/checkout-sidebar-jetpack-2-year-plans && yarn start`
- Go to https://wordpress.com/checkout/jetpack/jetpack_backup_t1_yearly
- Open the billing-term variant dropdown and verify you see the "2 year" option and the intro pricing text says, "$143.05 for first 2 years then $238.80 per 2 years". (See screenshot)
- With the "One year" option selected, verify you see the sidebar billing-term upsell and that it makes sense. (See screenshot)
- Click the upsell CTA ("Switch to a two year plan"), and verify the 2 year variant gets selected.
- Perform the same above checks for https://wordpress.com/checkout/jetpack/jetpack_security_t1_yearly

### Screenshots

**BEFORE** | **AFTER**
--- | ---
Markup 2023-05-25 at 09 35 34 | Markup 2023-05-25 at 09 33 41

--------

**BEFORE** | **AFTER**
--- | ---
Markup 2023-05-25 at 09 39 55 | Markup 2023-05-25 at 09 41 47

---------

**WPCOM Upselll comparison**

Markup 2023-05-25 at 09 46 00

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by sirbrillig** [comment]
I think this was already handled, but just out of curiosity, what does this change to the variant subtext look like at mobile width?

**Comment by elliottprogrammer** [comment]
@sirbrillig, thanks for spotting that. I fixed the intro pricing text so that on mobile, 2 and 3 year intro offers (or more) display the short version of text, same as the yearly (1 year) intro offers.
Now it will look like this:

Markup 2023-05-25 at 13 09 00

**Comment by sirbrillig** [comment]
Nice. Yeah, there's always two things I think to look for when making changes to the Jetpack variant picker: how does it affect WPCom plans (not at all in this case I think because it's forking the logic), and how does it look on mobile.

File: client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx [hunk 1]
```diff
@@ -99,17 +107,36 @@ export function CheckoutSidebarPlanUpsell() {
 		{ strong: createElement( 'strong' ) }
 	);
 
+	const twoYearCostLabel = createInterpolateElement( __( '<strong>2 Year Cost</strong>' ), {
+		strong: createElement( 'strong' ),
+	} );
```
Comment (robertsreberski) [inline client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx]: hmm, in this case, that we wrap the whole text in <strong /> - cannot we just nest this translation in HTML like `<strong>{ __( '2 Year Cost' ) }</strong>`

File: client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx [hunk 2]
```diff
@@ -99,17 +107,36 @@ export function CheckoutSidebarPlanUpsell() {
...
+								currentVariant.currency,
+								{
+									stripZeros: true,
+									isSmallestUnit: true,
+								}
+						  )
+						: formatCurrency( currentVariant.priceInteger, currentVariant.currency, {
+								stripZeros: true,
+								isSmallestUnit: true,
+						  } ) }
```
Comment (robertsreberski) [inline client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx]: Looks like the only difference between these cases is the currency value passed. We probably could simplify it a bit:

```
formatCurrency(
								currentVariant.priceInteger + ( isComparisonWithIntroOffer ? currentVariant.priceBeforeDiscounts : 0 ),
								currentVariant.currency,
								{
									stripZeros: true,
									isSmallestUnit: true,
								}
						  )
```

**Review by robertsreberski — CHANGES_REQUESTED** [review]
Code tests well in general, I left minor comments regarding code readability. Regarding the dropdown - I would ensure there are no widows (single word in the last line):

CleanShot 2023-05-25 at 19 59 28@2x

File: client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx (hunk 1 above)
Comment (elliottprogrammer) [reply client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx]: Ahh yes, totally. Good point, thank you. 👍

File: client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx (hunk 2 above)
Comment (elliottprogrammer) [reply client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx]: Ahh yes, this is great way to simplify. Thanks Robert! 🙌

File: client/my-sites/checkout/composite-checkout/components/item-variation-picker/variant-dropdown-price.tsx [hunk 3]
```diff
@@ -206,13 +207,17 @@ export const ItemVariantDropDownPrice: FunctionComponent< {
...
+							)
 					  )
-					: translate(
-							'%(formattedCurrentPrice)s for first %(introCount)s years then %(formattedPriceBeforeDiscounts)s per %(billingTermInYears)s years',
-							{ args }
+					: preventWidows(
+							translate(
+								'%(formattedCurrentPrice)s for first %(introCount)s years then %(formattedPriceBeforeDiscounts)s per %(billingTermInYears)s years',
+								{ args }
+							)
```
Comment (robertsreberski) [inline client/my-sites/checkout/composite-checkout/components/item-variation-picker/variant-dropdown-price.tsx]: I think we could wrap the whole expression altogether with `preventWidows()` but I'm also okay with keeping it like this. Great to reuse this function btw!

**Review by robertsreberski — APPROVED** [review]
The code LGTM to me now!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8011564

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8011564).

**Pull request merged**